openc3/lib/openc3/interfaces/protocols/fixed_protocol.rb

```diff
@@ -136,7 +136,13 @@ def identify_and_finish_packet
           @packet_name = identified_packet.packet_name
 
           # Get the data from this packet
-          packet_data = @data.slice!(0, identified_packet.defined_length + @discard_leading_bytes)
+          # Previous implementation looked like the following:
+          #   packet_data = @data.slice!(0, identified_packet.defined_length + @discard_leading_bytes)
+          # But slice! is 6x slower at small packets (1024)
+          # and 1000x slower at large packets (1Mb)
+          # Run test/benchmarks/string_mod_benchmark.rb for details
+          packet_data = @data[0...identified_packet.defined_length + @discard_leading_bytes]
+          @data = @data[(identified_packet.defined_length + @discard_leading_bytes)..-1]
```

Comment:
I think this is equivalent functionality. I tried to add a new test case to the spec but not sure if I fully covered the case because if you change `..-1` to `..-2` it still passes